app/assets/javascripts/angular-libs/ngs3upload.js

```diff
@@ -203,60 +208,67 @@ angular.module('ngS3upload.directives', []).
+                        //   scope.s3UploadObject = ngModel.$viewValue;
+                        //   scope.$parent[obj] = ngModel.$viewValue;
+                        // } else {
+                        //   scope.filename = ngModel.$viewValue;
+                        // }
+                        ngModel.$setValidity('uploading', true);
+                        ngModel.$setValidity('succeeded', true);
+                      }, function (resp) {
+                        alert('There was an error uploading your image. ' +
+                              'Developers: See JS console for details.');
+                        console.log('Bad response from Amazon S3:');
+                        console.log(resp);
+                        scope.fileUrl = resp.responseXML
                           .getElementsByTagName('Location')[0].innerHTML;
-                      if (attrs.s3Guide){
-                        guide = scope.$parent[attrs.s3Guide];
-                      }
-                      if (attrs.s3Stage){
-                        stage = guide.selectedStages[attrs.s3Stage];
-                        stage.featured_image = location;
-                      } else if (guide){
-                        guide.featured_image = location;
-                      }
-                      // TODO: Un-hardcode this.
-                      ngModel.$setViewValue(s3Uri + key);
-                      if (attrs.s3UploadObject){
-                        // TODO: make sure that guide and stage code mess from
-                        // above conform to this.
-                        var obj = attrs.s3UploadObject;
-                        scope.$parent[obj] = ngModel.$viewValue;
-                      } else {
                         scope.filename = ngModel.$viewValue;
-                      }
-                      ngModel.$setValidity('uploading', true);
-                      ngModel.$setValidity('succeeded', true);
-                    }, function (resp) {
-                      alert("There was an error uploading your image. Developers: See JS console for details.");
-                      console.log('Bad response from Amazon S3:');
-                      console.log(resp);
-                      scope.fileUrl = resp.responseXML
-                        .getElementsByTagName('Location')[0].innerHTML;
-                      scope.filename = ngModel.$viewValue;
-                      ngModel.$setValidity('uploading', true);
-                      ngModel.$setValidity('succeeded', false);
-                    });
+                        ngModel.$setValidity('uploading', true);
+                        ngModel.$setValidity('succeeded', false);
+                      });
 
-                }, function (error) {
-                  throw Error("Can't receive the needed options for S3 " + error);
-                });
+                  }, function (error) {
+                    throw Error("Can't receive the needed options for S3 " + error);
```

Comment:
Line is too long.<br>Mixed double and single quotes.<br>Missing 'new' prefix when invoking a constructor.